FTB/Signatures/CrashInfo.py

```diff
@@ -232,6 +232,9 @@ def fromRawCrashData(stdout, stderr, configuration, auxCrashData=None, cacheObje
             elif minidumpFirstDetected and minidumpSecondString in line:
                 result = MinidumpCrashInfo(stdout, stderr, configuration, auxCrashData)
                 break
+            elif line.startswith("==") and re.match(ValgrindCrashInfo.MSG_REGEX, line):
```

Comment:
Would it make sense here to instead just check for the header of the Valgrind output itself, e.g. look for "== Using Valgrind" or "== Memcheck, ..."? The purpose of that code in fromRawCrashData is solely to decide somewhat reliably what crash info it is dealing with. For Valgrind traces (and potentially some other formats which output the ==PID== prefix), this will increase load because a lot of lines will have to be matched for that regex before finally finding some matching error message.

Reply:
"== Using Valgrind" or "== Memcheck, ..." are not always in the log. For example if Valgrind is run with -q you will only see the summary I have in most of the test cases. The check for "==" is in place to prevent running the regex more than necessary (when we can avoid it). If other tools using the "==PID==" prefix are supported we should detect them before scanning the entire log. Worst case is if we have a large log were all lines start with  "==" and no reports are detected, which I think could happen but will be fairly uncommon

Reply:
I see, thanks. We should make sure that we run with -q then to avoid scanning whole logs with regex. Because if we had to scan through that whole Valgrind header and various other output with regex on every submit before finding the first report, this could considerably affect the server load when receiving lots of these logs. If we use -q though and there is really only the report itself, then that should be fine.